components/block/element.js

```diff
@@ -9,56 +9,93 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    blockId: {type: String, attribute: 'data-ngl-block-id'},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
-  }
+  constructor() {
+    super();
 
-  edit(e){
-    console.log(e.target.innerText)
-    this.refresh();
+    this.isSaveListenerAttached = false;
   }
 
-  delete(e) {
-    console.log(e.target.innerText)
-    this.remove();
+  getBlock() {
+    return window.parent.Core.g.layout.blocks.findWhere({id: this.blockId});
   }
 
-  moveUp(e) {
-    console.log(e.target.innerText)
-    const prevElement = this.previousElementSibling;
-    isBlock(prevElement) && this.parentNode.insertBefore(this, prevElement)
+  renderMenu() {
+    const block = this.getBlock();
+    const isInLinkedZone = block.zone().is_linked();
+
+    if (isInLinkedZone) {
+      return html`
+        <div class="edit-menu">
+          <button @click=${this.refresh}>Refresh</button>
+        </div>
+      `;
+    } else {
+      const parentId = block.attributes.parent_block_id;
+      const parent = document.querySelector(
+        `ngl-block[data-ngl-block-id="${parentId}"]`
+      );
+
+      if (parent && isBlock(parent)) {
+        return html`
+          <div class="edit-menu">
+            <button @click=${parent.edit.bind(parent)}>Edit container</button>
+            <button @click=${parent.refresh.bind(parent)}>
+              Refresh container
+            </button>
+            <button @click=${this.edit}>Edit</button>
+            <button @click=${this.refresh}>Refresh</button>
+          </div>
+        `;
+      } else {
+        return html`
+          <div class="edit-menu">
+            <button @click=${this.edit}>Edit</button>
+            <button @click=${this.refresh}>Refresh</button>
+          </div>
+        `;
+      }
+    }
   }
 
-  moveDown(e) {
-    console.log(e.target.innerText)
-    const nextElement = this.nextElementSibling;
-    isBlock(nextElement) && this.parentNode.insertBefore(nextElement, this)
+  edit() {
+    const block = this.getBlock();
+    block.trigger('edit');
+
+    if (!this.isSaveListenerAttached) {
+      block.on('sidebar_save:success', () => {
+        this.refresh();
+      });
+      this.isSaveListenerAttached = true;
+    }
   }
 
   async fetch() {
-    const resp = await fetch(window.location.href)
-    return resp.text()
+    const resp = await fetch(window.location.href);
+    return resp.text();
+  }
+
+  get main() {
+    return this.shadowRoot.querySelector('main');
   }
 
   async refresh() {
+    this.main.classList.add('loading');
```

Comment:
instead of this create new static property

```js
 static properties = {
    loading: {state: true}
```

Now you can just do:
```js
  this.loading = true;
  let html = await this.fetch();
  this.loading = false;
```

or even better:
```js
 // Move it to fetch function
 async fetch() {
   this.loading = true;
   const resp = await fetch(window.location.href);
   this.loading = false;
   return resp.text();
  }

 // and here you can just have
 let html = await this.fetch();
```
than in main render use classMap:

https://lit.dev/docs/templates/directives/#classmap

```js
render() {
    const classes = { loading: this.loading };
    
    return html`
      <main  class=${classMap(classes)} >
        ${this.renderMenu()}
        <slot></slot>
      </main>
    `;
}
```

:warning:  don't forget to import classMap directive

Reply:
Fixed in f46ee955e776e30d4608ac5cc0b071cadb0f8ac8